src/ansys/fluent/core/utils/search.py

```diff
@@ -212,10 +251,230 @@ def inner(tree, path, root_path):
+                api_object_names.add(k)
+                if "tui" in next_path:
+                    api_tui_objects.append(f"{next_path} ({type_})")
+                else:
+                    api_objects.append(f"{next_path} ({type_})")
                 if _match(k, word, match_whole_word, match_case):
-                    type_ = "Object" if isinstance(v, Mapping) else v
                     print(f"{next_path} ({type_})")
             if isinstance(v, Mapping):
                 inner(v, next_path, root_path)
 
     inner(api_tree, "", root_path)
+
+    def _write_file(
+        api_objects: list,
+        text: Optional[bool] = False,
+        tui: Optional[bool] = False,
+        name: Optional[bool] = None,
+    ):
+        api_tree_file = get_api_tree_file_name(text=text, name=name, tui=tui)
+        api_tree_file.touch()
+        with open(api_tree_file, "w") as file:
+            for api_object in api_objects:
+                file.write(f"{api_object}")
+                file.write("\n")
+
+    if pyfluent.WRITE_API_SEARCH_OBJECTS_FILE:
+
+        text_file_folder = Path(os.path.join(_THIS_DIRNAME, "api_tree"))
+        text_file_folder.mkdir(parents=True, exist_ok=True)
+
+        _write_file(api_objects=api_objects, text=True)
+        _write_file(api_objects=api_tui_objects, tui=True)
+        _write_file(api_objects=api_object_names, name=True)
+
+
+def _process_wildcards(word: str, names: list):
+    """Process wildcard pattern in the given word.
+
+    Parameters
+    ----------
+    word: str
+        The word to search for.
+    names: list
+        All API object names.
+
+    Returns
+    -------
+    wildcard_matches: list
+        Matched API object names.
+    """
+    wildcard_matches = []
+    for name in names:
+        if fnmatch.fnmatch(name, word):
+            wildcard_matches.append(name)
+    return wildcard_matches
+
+
+def _process_misspelled(
+    word: str,
+    names_txt_file,
+    names: list,
+    match_whole_word: Optional[bool] = False,
+    match_case: Optional[bool] = False,
+):
+    """Process misspelled word.
+
+    Parameters
+    ----------
+    word: str
+        The word to search for.
+    names_txt_file:
+        Text file containing all API object names.
+    names: list
+        All API object names.
+    match_whole_word: bool
+        Whether to match whole case.
+    match_case: bool
+        Whether to match case.
+
+    Returns
+    -------
+    correct_spell: list
+        List of corrected spell.
+    """
+    correct_spell = []
+    possible_corrections = []
+    spell = SpellChecker()
+    spell.word_frequency.load_text_file(names_txt_file)
+    spell.word_frequency.load_words(names)
+    misspelled = spell.unknown([word])
+    if misspelled:
+        for name in misspelled:
+            correct_spell.append(spell.correction(name))
+            possible_corrections.extend(list(spell.candidates(name)))
+        if match_whole_word and correct_spell == possible_corrections:
```

Comment:
I don't see the significance of these two lists being identical. What does it mean, and how does it correlate with `match_whole_word`?

Reply:
`match_whole_word` is a single possibility then both `correction` and `candidates` should be same otherwise it will give results for `match_case`.